---
title: "Circos plot tools"
author: "Ankit Dutta and JB Alberge"
date: September 9, 2022
output: 
  html_document:
    toc: true
    toc_depth: 2
    df_print: kable
    keep_md: true
---

# Init session

When running this code from console, `setwd` to `2_wgs`.

```{r cache=FALSE, message=FALSE, warning=FALSE}

source("0_annotate_samples.R")
source("../0_utils/circos_utils.R")

dir.create("figures")

CNV.df <- read_tsv("data/Sept9_reprocessed.cnv.revisions3.seg") # copy number can be public
maf <- read_maf("../data/20221017_AllPairsAndAD_maf_ac_pp_ccf_fit_v3.maf") # mutations are too specific for public-access. needs dbGAP level
# SVs are split into IGH/MYC specific calls and genome-wide calls.
ctf.tx <- read_tsv("../data/CTF.whitelist.oct17.txt")
svs <- read_xlsx("../data/CMMCs_ConsensusSV_filtered_results_whitelist_Oct17.xlsx") 

contigs <- factor(1:22, levels=c(1:22))
sectors <- paste0("chr", contigs) # circlize uses UCSC chr notation.

```

# Post-process CNVs

`YMIN`/`YMAX` will set the offset for graphical representation 

```{r}
contigs <- factor(1:22, levels=c(1:22))

CNV.df <- CNV.df %>% 
  inner_join(clinicaldata, by=c("entity:sample_id")) %>% 
  filter(Chromosome %in% contigs) %>%
  mutate(Chromosome=factor(Chromosome, levels=contigs),
         Norm_Segment_Mean = pmax(0, pmin(Segment_Mean, 4)),
         YMIN=ifelse(tissue=="BMPCs", 0, 1), 
         YMAX =ifelse(tissue=="BMPCs", 1, 2))
```

# Post-process SNV and indels

WGS pipeline adds a `_tumor` to `pair_id` that we need to get rid of.

```{r}
maf <- maf |>
  mutate(sample_id=str_remove(Tumor_Sample_Barcode, "_tumor")) |>
  left_join(clinicaldata, by=c("sample_id"="entity:sample_id"))
```

# Post-process SVs

For manual review of IGH/MYC specific SVs and REBC-like SVs.

```{r}
ctf.tx <- ctf.tx %>% 
  filter(Keep=="TRUE") %>%
  left_join(clinicaldata, by=c("ID"="entity:sample_id")) %>%
  relocate(CHR.IG, IG_START, CHR.ONCO, PARTNER_START) %>% 
  mutate(chr1 = paste0("chr", CHR.IG), 
         chr2 = paste0("chr", CHR.ONCO),
         start1 = IG_START,
         end1 = IG_START,
         start2 = PARTNER_START,
         end2 = PARTNER_START) %>%
  mutate(chr1 = case_when(chr1=="chr23" ~ "ChrX",
                          chr1=="chr24" ~ "ChrY",
                          TRUE ~ chr1)) %>%
    group_by(ID, IG, ONCO) %>%
    slice_head(n=1) %>%
    filter(chr1 %in% sectors & chr2 %in% sectors)
svs <- svs %>% filter(keep==1) %>% left_join(clinicaldata, by=c("individual"="entity:sample_id"))
```


# Re-format all for circlize

## CTF025 and CTF031 examples

```{r}

p.id="CTF025_CMMCs_198"

plot.single.circos(p.id = p.id, sectors = sectors, ctf = ctf.tx, cn=CNV.df, svs=svs, maf=maf, non.synonymous = non.synonymous, genes.of.interest = genes.of.interest, filename = "figures/" %+% p.id)
plot.single.circos(p.id = p.id, sectors = sectors, ctf = ctf.tx, cn=CNV.df, svs=svs, maf=maf, non.synonymous = non.synonymous, genes.of.interest = genes.of.interest)
```



```{r}

p.id="CTF031_CMMCs_41500"

plot.single.circos(p.id = p.id, sectors = sectors, ctf = ctf.tx, cn=CNV.df, svs=svs, maf=maf, non.synonymous = non.synonymous, genes.of.interest = genes.of.interest, filename = "figures/" %+% p.id)
plot.single.circos(p.id = p.id, sectors = sectors, ctf = ctf.tx, cn=CNV.df, svs=svs, maf=maf, non.synonymous = non.synonymous, genes.of.interest = genes.of.interest)

```

## Chromothripsis of chr 3 in CTF033

```{r}

p.id="CTF033_CMMCs_755"

p1 <- plot.single.circos(p.id = p.id, sectors = sectors, ctf = ctf.tx, cn=CNV.df, svs=svs, maf=maf, non.synonymous = non.synonymous, genes.of.interest = genes.of.interest)
p2 <- plot.single.circos(p.id = p.id, sectors = "chr3", ctf = ctf.tx, cn=CNV.df, svs=svs, maf=maf, non.synonymous = non.synonymous, genes.of.interest = genes.of.interest, axis.labels.cex = .5, plotType = c('axis', 'ideogram', 'labels'))
plot_grid(p1, p2)

plot.single.circos(p.id = p.id, sectors = sectors, ctf = ctf.tx, cn=CNV.df, svs=svs, maf=maf, non.synonymous = non.synonymous, genes.of.interest = genes.of.interest, filename = "figures/" %+% p.id %+% ".pdf")
plot.single.circos(p.id = p.id, sectors = "chr3", ctf = ctf.tx, cn=CNV.df, svs=svs, maf=maf, non.synonymous = non.synonymous, genes.of.interest = genes.of.interest, filename = "figures/" %+% p.id %+% "_zoom.pdf", axis.labels.cex = .5, plotType = c('axis', 'ideogram', 'labels'))


```

## Chromoplexy of chr 7, 8, 18 in CTF034

### In CMMCs:

```{r}

p.id="CTF034_CMMCs_654"
sectors.ctf034 <- "chr" %+% c(7, 8, 18)

plot.single.circos(p.id = p.id, sectors = sectors.ctf034, ctf = ctf.tx, cn=CNV.df, svs=svs, maf=maf, non.synonymous = non.synonymous, genes.of.interest = genes.of.interest, axis.labels.cex = .5, plotType = c('axis', 'ideogram', 'labels'))

plot.single.circos(p.id = p.id, sectors = sectors.ctf034, ctf = ctf.tx, cn=CNV.df, svs=svs, maf=maf, non.synonymous = non.synonymous, genes.of.interest = genes.of.interest, axis.labels.cex = .5, plotType = c('axis', 'ideogram', 'labels'), filename = "figures/" %+% p.id %+% "_zoom.pdf")

```

### In BMPCs:

```{r}
p.id="CTF034_BMPCs_16800"
sectors.ctf034 <- "chr" %+% c(7, 8, 18)

plot.single.circos(p.id = p.id, sectors = sectors.ctf034, ctf = ctf.tx, cn=CNV.df, svs=svs, maf=maf, non.synonymous = non.synonymous, genes.of.interest = genes.of.interest, axis.labels.cex = .5, plotType = c('axis', 'ideogram', 'labels'))

plot.single.circos(p.id = p.id, sectors = sectors.ctf034, ctf = ctf.tx, cn=CNV.df, svs=svs, maf=maf, non.synonymous = non.synonymous, genes.of.interest = genes.of.interest, axis.labels.cex = .5, plotType = c('axis', 'ideogram', 'labels'), filename = "figures/" %+% p.id %+% "_zoom.pdf")
```


## CTF047 example

```{r}

p.id="CTF047_BMPCs"

plot.single.circos(p.id = p.id, sectors = sectors, ctf = ctf.tx, cn=CNV.df, svs=svs, maf=maf, non.synonymous = non.synonymous, genes.of.interest = genes.of.interest, filename = "figures/CTF047_PBMCs.pdf")
plot.single.circos(p.id = p.id, sectors = sectors, ctf = ctf.tx, cn=CNV.df, svs=svs, maf=maf, non.synonymous = non.synonymous, genes.of.interest = genes.of.interest)

p.id="CTF047_CMMCs_75k"

plot.single.circos(p.id = p.id, sectors = sectors, ctf = ctf.tx, cn=CNV.df, svs=svs, maf=maf, non.synonymous = non.synonymous, genes.of.interest = genes.of.interest, filename = "figures/" %+% p.id %+% ".pdf")
plot.single.circos(p.id = p.id, sectors = sectors, ctf = ctf.tx, cn=CNV.df, svs=svs, maf=maf, non.synonymous = non.synonymous, genes.of.interest = genes.of.interest)
```

## CTF032 over time

```{r}

p.id="CTF032_CMMCs_9k"

plot.single.circos(p.id = p.id, sectors = sectors, ctf = ctf.tx, cn=CNV.df, svs=svs, maf=maf, non.synonymous = non.synonymous, genes.of.interest = genes.of.interest, filename = "figures/" %+% p.id %+% ".pdf")
plot.single.circos(p.id = p.id, sectors = sectors, ctf = ctf.tx, cn=CNV.df, svs=svs, maf=maf, non.synonymous = non.synonymous, genes.of.interest = genes.of.interest)

p.id="CTF032_CMMCs_T1_2200"

plot.single.circos(p.id = p.id, sectors = sectors, ctf = ctf.tx, cn=CNV.df, svs=svs, maf=maf, non.synonymous = non.synonymous, genes.of.interest = genes.of.interest, filename = "figures/" %+% p.id %+% ".pdf")
plot.single.circos(p.id = p.id, sectors = sectors, ctf = ctf.tx, cn=CNV.df, svs=svs, maf=maf, non.synonymous = non.synonymous, genes.of.interest = genes.of.interest)
```

# Session info

```{r, eval=TRUE, include=TRUE, comment=NA}
sessionInfo()
```
